src/price-formatting/parse.js

```diff
@@ -96,84 +96,100 @@ function parseDecimalprice(numberFormatting, s, formatFlags) {
     return 0;
 }
 
-function parseFractionalPrice(numberFormatting, s, formatFlags, decimals) {
-
+function parseModernFractionalPrice(numberFormatting, s, decimals) {
     let result;
+    const separator = getModernFractionsSeparator(numberFormatting);
+    const denominator = 1 << decimals;
+
+    const pipIndex = s.indexOf(separator);
+    if (pipIndex !== -1) {
+        const integerPart = s.substring(0, pipIndex).trim();
+        if (integerPart.length > 0) {
+            result = numberFormatting.parse(integerPart);
+        } else {
+            result = 0;
+        }
 
-    if (formatFlags.ModernFractions) {
-        // special futures
-
-        const separator = getModernFractionsSeparator(numberFormatting);
-        const denominator = 1 << decimals;
-
-        const pipIndex = s.indexOf(separator);
-        if (pipIndex !== -1) {
-            const integerPart = s.substring(0, pipIndex).trim();
-            if (integerPart.length > 0) {
-                result = numberFormatting.parse(integerPart);
-            } else {
-                result = 0;
-            }
-
-            if (pipIndex + 1 < s.length) {
-                const pipPart = numberFormatting.parse(s.substring(pipIndex + 1).trim());
+        if (pipIndex + 1 < s.length) {
+            const pipPart = numberFormatting.parse(s.substring(pipIndex + 1).trim());
 
-                if (pipPart < denominator) {
-                    result = result += (pipPart / denominator);
+            if (pipPart < denominator) {
+                // checking of "-0" in integerPart
+                if (integerPart.length === 2 &&
+                    integerPart.substr(0, 1) === '-' &&
+                    integerPart.substring(1) === '0') {
+                    result = -1 * (pipPart / denominator);
                 } else {
-                    result = NaN;
+                    result = result += (pipPart / denominator);

```

Comment:
I think it can just be:
```result += (pipPart / denominator);```

No need for ```result = ```

Reply:
Done